packages/ngx-schematics/src/page/schema.json

```diff
@@ -0,0 +1,57 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "SchematicsPage",
+  "title": "Page Options Schema",
+  "type": "object",
+  "properties": {
+    "path": {
+      "type": "string",
+      "format": "path",
+      "$default": {
+        "$source": "workingDirectory"
+      },
+      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
+      "visible": false
+    },
+    "project": {
+      "type": "string",
+      "description": "The name of the project.",
+      "$default": {
+        "$source": "projectName"
+      }
+    },
+    "name": {
+      "type": "string",
+      "description": "The name of the page.",
+      "$default": {
+        "$source": "argv",
+        "index": 0
+      },
+      "x-prompt": "What name would you like to use for the page?"
+    },
+    "pageId": {
+      "type": "string",
+      "description": "The id of the action used to access the page.",
+      "x-prompt": "What is the page id?"
+    },
+    "iconClass": {
+      "type": "string",
+      "description": "The class for the icon displayed on the page.",
+      "x-prompt": "What is the iconClass?"
+    },
+    "style": {
+      "description": "The file extension or preprocessor to use for style files, or 'none' to skip generating the style file.",
+      "type": "string",
+      "default": "less",
+      "enum": ["css", "scss", "sass", "less", "none"]
+    },
+    "entrypoint": {
+      "description": "Defines how the page will be available to the user.",
+      "type": "string",
+      "default": "Menu Item",
+      "enum": ["Menu Item", "Action Button"],
+      "x-prompt": "How will the page be accessible?"
+    }
+  },
+  "required": ["name", "pageId", "iconClass"]
```

Comment:
Can the iconClass be made optional? You could use the generic icon if not specified: icon-core-et-sm-generic

Reply:
Yeah, we could do it that way. It may be the case that the page needs a new icon and the developer does not have it yet, so yeah we could add a default.